src/vcpkg/commands.extract.cpp

```diff
@@ -0,0 +1,166 @@
+#include <vcpkg/base/strings.h>
+
+#include <vcpkg/archives.h>
+#include <vcpkg/commands.extract.h>
+#include <vcpkg/commands.h>
+#include <vcpkg/vcpkgcmdarguments.h>
+#include <vcpkg/vcpkgpaths.h>
+
+namespace vcpkg::Commands
+{
+    static constexpr StringLiteral OPTION_STRIP = "strip";
+
+    static constexpr std::array<CommandSetting, 1> EXTRACT_SETTINGS = {{
+        {OPTION_STRIP, []() { return msg::format(msgStripOption, msg::option = "strip"); }},
+    }};
+
+    const CommandStructure ExtractCommandStructure = {
+        [] { return msg::format(msgExtractHelp); },
+        2,
+        3,
+        {{}, {EXTRACT_SETTINGS}, {}},
+        nullptr,
+    };
+
+    static StripSetting get_strip_setting(const ParsedArguments& options)
+    {
+        auto iter = options.settings.find(OPTION_STRIP);
+        if (iter != options.settings.end())
+        {
+            std::string maybe_value = iter->second;
+
+            if (Strings::case_insensitive_ascii_equals(maybe_value, "auto"))
+            {
+                return {StripMode::Automatic, -1};
+            }
+
+            auto maybe_strip_value = Strings::strto<int>(maybe_value);
+            if (auto value = maybe_strip_value.get(); value && *value >= 0)
+            {
+                return {StripMode::Manual, *value};
+            }
+
+            // If the value is not an integer or is less than 0
+            msg::println_error(msgErrorInvalidExtractOption, msg::option = OPTION_STRIP, msg::value = maybe_value);
+            Checks::exit_fail(VCPKG_LINE_INFO);
+        }
+        // No --strip set, default to 0
+        return {StripMode::Manual, 0};
+    }
+
+    constexpr IsSlash is_slash;
+    size_t get_common_directories_count(std::vector<Path> paths)
+    {
+        if (paths.size() == 0)
+        {
+            return 0;
+        }
+
+        std::string known_common_prefix = paths[0].native();
+        for (size_t idx = 1; idx < paths.size(); ++idx)
+        {
+            auto&& candidate = paths[idx].native();
+            auto mismatch_point = std::mismatch(known_common_prefix.begin(),
+                                                known_common_prefix.end(),
+
+                                                candidate.begin(),
+                                                candidate.end())
+                                      .first;
+            known_common_prefix.erase(mismatch_point, known_common_prefix.end());
+        }
+
+        return std::count_if(known_common_prefix.begin(), known_common_prefix.end(), is_slash);
+    }
+
+    std::vector<std::pair<Path, Path>> get_archive_deploy_operations(const ExtractedArchive& archive,
+                                                                     StripSetting strip_setting)
+    {
+        std::vector<std::pair<Path, Path>> result;
+
+        const auto temp_dir = archive.temp_path;
+        const auto base_path = archive.base_path;
+        const auto proximate = archive.proximate_to_temp;
```

Comment:
```suggestion
        const auto& temp_dir = archive.temp_path;
        const auto& base_path = archive.base_path;
        const auto& proximate = archive.proximate_to_temp;
```

Are those copies necessary?